src/nft-card.ts

```diff
@@ -104,30 +121,37 @@ export class NftCard extends LitElement {
         await this.requestUpdate()
 
         // Watch for the account to change then update state of component
-        window.ethereum.on('accountsChanged', function (accounts) {
-            [this.account] = accounts
-            console.log('accountsChanged', accounts)
-        })
+        // window.ethereum.on('accountsChanged',  (accounts: Array<string>) => {
+        //     this.account = accounts.length > 0 ? accounts[0] : ''
+        //     console.log('accountsChanged', accounts)
+        // })
     }
-
     public async buyEvent() {
+        // Check to see that we are connected and we have an account
+        // This will run if user is logged out or has not granted access
+        if (this.provider) {
+            await this.connectWallet()
+            this.account = this.provider ? this.provider.selectedAddress : null
+        }
+        if (this.isUnlocked) {
+            const order = await this.seaport.api.getOrder({
```

Comment:
I think we should use the first order inside `this.asset.sellOrders` to save a get request